rocksdb_replicator/thrift/replicator.thrift

```diff
@@ -60,6 +60,7 @@ enum ErrorCode {
   OTHER = 0,
   SOURCE_NOT_FOUND = 1, # could not find the upstream db
   SOURCE_READ_ERROR = 2,
+  SOURCE_REMOVED = 3, # upstream db was recently removed
```

Comment:
Can you please add some comments that this is an optimization to reduce the helix reads? And it's no harm if the upstream fails to differentiate between SOURCE_NOT_FOUND VS. SOURCE_REMOVED.